File: 0061-rotate-list/0061-rotate-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode rotateRight(ListNode head, int k) {
        if (head == null || head.next == null || k == 0) {
            return head;
        }

        // Count the length of the list
        ListNode temp = head;
        int count = 1;
        while (temp.next != null) {
            count++;
            temp = temp.next;
        }

        // Make it circular
        temp.next = head;

        // Normalize k
        k = k % count;
        k = count - k;

        // Move to the (k)th node
        while (k-- > 0) {
            temp = temp.next;
        }

        // Break the circle
        ListNode newHead = temp.next;
        temp.next = null;

        return newHead;
    }
}
        

File: 0061-rotate-list/0061-rotate-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def rotateRight(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
        if k==0 or head==None:
            return head
        temp=head
        count=1
        while temp.next:
            temp=temp.next
            count+=1
        temp.next=head
        k=k%count
        k=count-k
        t=head
        for i in range(k-1):
            t=t.next
        newhead=t.next
        t.next=None
        return newhead
            
        